### PR TITLE
[aom] Add new port

### PR DESCRIPTION
**Pull request by HappySeaFox**
**Describe the pull request**

Add AV1 codec library https://aomedia.googlesource.com/aom version 3.1.0.

- #### What does your PR fix?  

\-

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

Triplets: Everything except UWP and ARM.

CI baseline is not updated.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  

Yes.

Port peculiarity:

- Added missing CMake rules to install and export unofficial targets. Discussion is in #17332 .
- AOM always builds an extra static lib when `BUILD_SHARED_LIBS` is `ON` which is not installed by the rules added above. We simply don't need it.

### CODE REVIEW

File: ports/aom/aom-install.diff [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ setup_aom_dsp_targets()
+ setup_aom_mem_targets()
+diff --git a/cmake/aom-config.cmake.in b/cmake/aom-config.cmake.in
+new file mode 100644
+index 000000000..91cac3b5b
+--- /dev/null
++++ b/cmake/aom-config.cmake.in
+@@ -0,0 +1,2 @@
++@PACKAGE_INIT@
++include(${CMAKE_CURRENT_LIST_DIR}/unofficial-aom-targets.cmake)
```
Comment (NancyLi1013) [inline ports/aom/aom-install.diff:75]: Can you submit these changes to upstream?

File: ports/aom/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "aom",
+  "version-semver": "3.1.0",
+  "description": "AV1 codec library",
+  "homepage": "https://aomedia.googlesource.com/aom",
+  "supports": "!uwp & !arm64",
```
Comment (NancyLi1013) [inline ports/aom/vcpkg.json:6]: Can you help confirm whether this port doesn't support `arm `or `arm64`? Since I noticed that you added `arm ` to the unsupported message in portfile.cmake.

File: ports/aom/aom-install.diff (hunk 1 above)
Comment (happy-sea-fox) [reply ports/aom/aom-install.diff:75]: Submitted a month ago with no progress so far. As per the suggestion in the previous PR, I create unofficial targets.

Comment (NancyLi1013) [reply ports/aom/aom-install.diff:75]: Thanks for your reply. If possible, please help add the PR link in this PR. Such as add it in  `PATCH aom-install.diff`. If this has been accepted by upstream, we can remove the patch.

File: ports/aom/vcpkg.json (hunk 2 above)
Comment (happy-sea-fox) [reply ports/aom/vcpkg.json]: Updated to `arm`, thanks!

File: ports/aom/aom-install.diff (hunk 1 above)
Comment (happy-sea-fox) [reply ports/aom/aom-install.diff:75]: Added a comment in portfile.cmake, thanks!

File: ports/aom/usage [hunk 3]
```diff
@@ -0,0 +1,4 @@
+The package aom provides CMake targets:
```
Comment (ras0219-msft) [inline ports/aom/usage:1]: Why is this file needed? Do vcpkg's heuristics not suggest the correct usage?

**Review by ras0219-msft — APPROVED** [review]
Thanks for the PR!

LGTM once my comment is answered :)

File: ports/aom/usage (hunk 3 above)
Comment (happy-sea-fox) [reply ports/aom/usage]: Thanks, updated! Usage is not needed really.

File: ports/aom/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    TAG v3.1.0
+    PATCHES
+        aom-rename-static.diff
+        # Can be dropped when https://bugs.chromium.org/p/aomedia/issues/detail?id=3029 is merged into the upstream
+        aom-install.diff
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (NancyLi1013) [inline ports/aom/portfile.cmake:26]:
```suggestion
```https://github.com/microsoft/vcpkg/blob/master/docs/users/versioning.md
This argument is not needed here. Please see the reference here.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by happy-sea-fox** [comment]
Ping!

**Comment by strega-nil-ms** [comment]
Thanks @happy-sea-fox :)
